src/riak_test_escript.erl

```diff
@@ -236,8 +236,54 @@ parse_command_line_tests(ParsedArgs) ->
         end, [], lists:usort(DirTests ++ SpecificTests)).
 
 extract_test_names(Test, {CodePaths, TestNames}) ->
-    {[filename:dirname(Test) | CodePaths],
-     [filename:rootname(filename:basename(Test)) | TestNames]}.
+
+    RTTestPaths = get_test_paths(),
+    CommaSepTests = string:tokens(Test, [$,]),
+
+    WCTestsNoPrefix = [get_test_with_valid_prefix(TestA, RTTestPaths) || TestA <- CommaSepTests, get_test_type(TestA) =:= yes_wc_no_prefix],
+    WCTestsPrefix = [TestA || TestA <- CommaSepTests, get_test_type(TestA) =:= yes_wc_yes_prefix],
+    NotWCTestsNoPrefix = [get_test_with_valid_prefix(TestA, RTTestPaths) || TestA <- CommaSepTests, get_test_type(TestA) =:= no_wc_no_prefix],
+    NotWCTestsPrefix = [TestA || TestA <- CommaSepTests, get_test_type(TestA) =:= no_wc_yes_prefix],
+
+    AllWCTests = lists:append([filelib:wildcard(TestC) || TestC <- WCTestsNoPrefix++WCTestsPrefix]),
+    AllNotWCTests = NotWCTestsNoPrefix++NotWCTestsPrefix,
+    AllTests = AllWCTests++AllNotWCTests,
+
+    CodePathList = [filename:dirname(TestName) || TestName <- AllTests],
```

Comment:
We can probably combine this so it only passes over the tests once.  Actually we might be able to build a list of tuples of code path and test and `lists:unzip/1` them when it's time to return the values.